packages/expo-router/ios/LinkPreview/LinkPreviewNativeView.swift

```diff
@@ -36,7 +36,7 @@ class NativeLinkPreviewView: ExpoView, UIContextMenuInteractionDelegate, LinkPre
   }
 
   // MARK: - Children
-
+#if RCT_NEW_ARCH_ENABLED
```

Comment:
quick question: since we have the `areNativeViewsAvailable` from js side, do we still need the check here? asking this because i would prefer to reduce the build time check as much as possible. it blocks prebuild (precompile xcframework).

also react-native core had the effort to reduce the new/legacy arch build time flag in the past, though we didn't get a chance to do that. i hope we can stop adding more code for that.